projects/app/src/pages/api/core/app/update.ts

```diff
@@ -24,39 +36,153 @@ async function handler(req: NextApiRequest, res: NextApiResponse<any>) {
     teamTags,
     defaultPermission
   } = req.body as AppUpdateParams;
+
   const { appId } = req.query as { appId: string };
 
   if (!appId) {
-    throw new Error('appId is empty');
+    Promise.reject(CommonErrEnum.missingParams);
   }
 
-  // 凭证校验
-  if (defaultPermission) {
-    await authApp({ req, authToken: true, appId, per: ManagePermissionVal });
-  } else {
-    await authApp({ req, authToken: true, appId, per: WritePermissionVal });
-  }
+  const app = await (async () => {
+    if (defaultPermission) {
+      // if defaultPermission or inheritPermission is set, then need manage permission
+      return (await authApp({ req, authToken: true, appId, per: ManagePermissionVal })).app;
+    } else {
+      return (await authApp({ req, authToken: true, appId, per: WritePermissionVal })).app;
+    }
+  })();
```

Comment:
const {app } = await (async () => {
    if (defaultPermission) {
      // if defaultPermission or inheritPermission is set, then need manage permission
      return authApp({ req, authToken: true, appId, per: ManagePermissionVal })
    } 

      return (await authApp({ req, authToken: true, appId, per: WritePermissionVal })).app;
  
  })();